spec-tests/src/test/java/org/candlepin/spec/owners/OwnerResourceSpecTest.java

```diff
@@ -792,4 +795,41 @@ public void shouldAllowUserWithOwnerPoolsPermissionCanSeeSystemPurposeOfTheOwner
         assertEquals(owner.getKey(), attributes.getOwner().getKey());
     }
 
+    @Test
+    @OnlyInHosted
+    public void shouldMigrateAllConsumersOfAnonymousOwnerToTargetOwner() {
+        String accountId = StringUtil.random("cloud-account-id-");
+        String instanceId = StringUtil.random("cloud-instance-id-");
+        String offerId = StringUtil.random("cloud-offer-");
+        OwnerDTO anonOwner = admin.owners().createOwner(Owners.random().anonymous(true));
+        OwnerDTO destOwner = owners.createOwner(Owners.random());
+        admin.hosted().createOwner(anonOwner);
+        ProductDTO prod = admin.ownerProducts().createProduct(anonOwner.getKey(), Products.random());
+        admin.hosted().createProduct(prod);
+        admin.owners().createPool(anonOwner.getKey(), Pools.random(prod));
+        admin.hosted().createSubscription(Subscriptions.random(anonOwner, prod));
+
+        admin.hosted().associateProductIdsToCloudOffer(offerId, List.of(prod.getId()));
+        admin.hosted().associateOwnerToCloudAccount(accountId, anonOwner.getKey());
+        CloudAuthenticationResultDTO result = ApiClients.noAuth().cloudAuthorization()
+            .cloudAuthorizeV2(accountId, instanceId, offerId, "test-type", "");
+
+        ApiClient anonClient = ApiClients.bearerToken(result.getToken());
+        anonClient.consumers().createConsumer(Consumers.random(anonOwner));
+        anonClient.consumers().createConsumer(Consumers.random(anonOwner));
+        anonClient.consumers().createConsumer(Consumers.random(anonOwner));
+
+        AsyncJobStatusDTO job = owners.claim(anonOwner.getKey(),
+            new ClaimantOwner().claimantOwnerKey(destOwner.getKey()));
```

Comment:
once the remaining work mentioned in other comments is done, we should assert after this call that the anonOwner has claimed==true, and claimant_owner==destOwner.key